app/src/main/java/org/luxoft/sdl_core/SdlLauncherService.java

```diff
@@ -76,25 +80,33 @@ public void run() {
             public void run() {
                 final Intent start_intent = new Intent(ON_SDL_SERVICE_STARTED);
                 getApplicationContext().sendBroadcast(start_intent);
+                updateServiceNotification("SDL core is running");
             }
         }, SDL_STARTED_DELAY);
 
         return super.onStartCommand(intent, flags, startId);
     }
     @Override
     public void onDestroy() {
+//        updateServiceNotification("SDL core is stopping...");
+
         try {
             StopSDLNative();
             if(sdl_thread_.isAlive()) {
                 sdl_thread_.join(SDL_JOIN_WAIT);
             }
+            if (sdl_thread_.isAlive()) {
+                sdl_thread_.interrupt();
+            }
         } catch (InterruptedException e) {
             e.printStackTrace();
         }
 
         final Intent intent = new Intent(ON_SDL_SERVICE_STOPPED);
         getApplicationContext().sendBroadcast(intent);
 
+        stopForeground(true);
+
```

Comment:
@AKalinich-Luxoft 
for a complete stop probably add stopSelf () :
https://stackoverflow.com/questions/20857120/what-is-the-proper-way-to-stop-a-service-running-as-foreground

  stopForeground(true);
  stopSelf()

Reply:
@VadymLuchko thank you :)
I used an approach from this comment - https://stackoverflow.com/a/50888586
Take a look at https://github.com/CustomSDL/sdl_android_launcher/pull/5/commits/b172d24a3aeffb2fe7cb183d547e11b090ea66f4

Reply:
hope this works

Reply:
@VadymLuchko it is. I checked :)